src/main.cpp

```diff
@@ -4081,11 +4074,6 @@ bool ContextualCheckBlockHeader(
 
     int nHeight = pindexPrev->nHeight+1;
 
-    // Check Equihash solution is valid
-    if (fCheckPOW && !CheckEquihashSolution(&block, nHeight, consensusParams))
```

Comment:
Since this has been moved from contextual to non-contextual checks, it will be possible that checks fail in a different location, does that have any negative effects if both 3.0.0 and 3.1.0 nodes are active on the network simultaneously?

Reply:
As far as I understand, no. Note that this is restoring how it was before 3.0.0.

Reply:
It shouldn't, because in both cases an invalid block header will never be accepted. This change is an exact revert of the prior commit that moved the checks from non-contextual to contextual; see https://github.com/zcash/zcash/pull/4448#discussion_r408704189 and https://github.com/zcash/zcash/pull/4451#issuecomment-613966669 for the similar reasoning for that change.